src/charm.py

```diff
@@ -64,6 +68,29 @@ def _on_config_changed(self, event) -> None:
         self.model.unit.status = MaintenanceStatus(msg)
         self.configure_workload_container(event)
 
+    def _report_visits_by_ip_action(self, event: ActionEvent) -> None:
+        """Handle the report-visits-by-ip action."""
+        results = self._report_visits_by_ip()
+        event.set_results({"ips": str(results)})
```

Comment:
Why wrapping wrap the map in a `ips` key?

Reply:
In that pastebin I see the `ips`key wrapping all the output. The output needs to be a map and not it already is

Reply:
sorry, wrong paste. That pastebin is the one with the ips wrap but results instead of str(results). when results is treated as a dict it prints like that. without the ips wrap (passing the dict as it is) it looks like this https://pastebin.canonical.com/p/gD4ffbZzyy/

Reply:
That certainly looks like a bug in the operator framework. Can you file one?

Reply:
Surely will :+1: